MapboxNavigation/PollyVoiceController.swift

```diff
@@ -117,7 +116,10 @@ public class PollyVoiceController: RouteVoiceController {
             input.voiceId = voiceId
         }
         
-        input.text = "<speak><amazon:effect name=\"drc\"><prosody volume='\(instructionVoiceVolume)' rate='\(instructionVoiceSpeedRate)'>\(text)</prosody></amazon:effect></speak>"
+        let removedClosingOpenTag = String(text.characters.dropFirst("<speak>".characters.count).dropLast("</speak>".characters.count))
```

Comment:
This removes the first seven and last eight characters, even if those characters aren’t `<speak>` and `</speak>`, respectively. That would lead to malformed XML if the Directions API happened to return something like `<speak version="1.0" xml:lang="en-US">`. (The `version` and `xml:lang` attributes are [required by the SSML specification](https://www.w3.org/TR/speech-synthesis/#S3.1.1), though Polly currently seems to be lenient about this requirement.)

Reply:
@1ec5 what would you suggest to change this to?

Reply:
The original code was slightly more robust; at least we could be sure that it was replacing `<speak>` and `</speak>`.

A better solution is to find and replace these regular expressions: `^<speak(?:\s.[^>]+)?>` and `</speak>$` (with whatever string escaping you need). This only works because we’re still making assumptions about what the API will give us. It’ll still fail to replace if the API gives us a valid XML document (with an XML prologue), but in that case we’ll just fail to apply the desired `<prosody>` tag; at least we won’t end up with a speech synthesis request that errors out.

Reply:
You could also future-proof that regular expression somewhat to account for `xml:lang` and `version` being supplied, as required by the SSML specification: replace `^<speak\s*(?:\s+[^>]*)?>` with `$0<amazon:effect …`